src/execution/operator/scan/physical_table_scan.cpp

```diff
@@ -27,14 +48,35 @@ void PhysicalTableScan::GetChunkInternal(ClientContext &context, DataChunk &chun
 		table.InitializeScan(transaction, state->scan_offset, column_ids);
 		state->initialized = true;
 	}
-
-	table.Scan(transaction, chunk, state->scan_offset);
+	//! Get max value
+    idx_t result_count = -1;
+    do{
+        chunk.Reset();
+        table.Scan(transaction, chunk, state->scan_offset, table_filters);
+        if (expression){
+            SelectionVector sel(STANDARD_VECTOR_SIZE);
+            idx_t initial_count = chunk.size();
+            result_count = state->executor.SelectExpression(chunk, sel);
```

Comment:
This looks like it's just the PhysicalFilter code duplicated here, which will not provide any performance boost. For now this is fine, but in the future this selection needs to be pushed into the actual scan (i.e. into the NumericSegment and StringSegment scans).

Reply:
True, I'll push that when fixing the strings next season